### PR TITLE
migrate training_multi-task-learning.py v2to v3

### PR DESCRIPTION
**Pull request by harshitsharma496**
model.fit is replaced with SentenceTransformerTrainer in docstrings

Before : model.fit(train_objectives=[(train_dataloader_MultipleNegativesRankingLoss, train_loss_MultipleNegativesRankingLoss), (train_dataloader_constrative_loss, train_loss_constrative_loss)] ...)

After : SentenceTransformerTrainer( train_dataset={ "mnrl": MultipleNegativesRankingLoss_train_dataset,"cl": ContrastiveLoss_train_dataset,.. },loss={"mnrl": MultipleNegativesRankingLoss(model=model),"cl": ContrastiveLoss(model=model),...})

### CODE REVIEW

**Comment by harshitsharma496** [comment]
For #3621

File: examples/sentence_transformer/training/quora_duplicate_questions/training_multi-task-learning.py [hunk 1]
```diff
@@ -7,8 +7,8 @@
 
 This script combines both losses to get the best of both worlds.
 
-Multi task learning is achieved quite easily by calling the model.fit method like this:
-model.fit(train_objectives=[(train_dataloader_MultipleNegativesRankingLoss, train_loss_MultipleNegativesRankingLoss), (train_dataloader_constrative_loss, train_loss_constrative_loss)] ...)
+Multi task learning is achieved quite easily by calling the SentenceTransformerTrainer method like this:
+SentenceTransformerTrainer(train_dataset={ "mnrl": MultipleNegativesRankingLoss_train_dataset,"cl": ContrastiveLoss_train_dataset},loss={"mnrl": MultipleNegativesRankingLoss(model=model),"cl": ContrastiveLoss(model=model)},...)
```
Comment (tomaarsen) [inline examples/sentence_transformer/training/quora_duplicate_questions/training_multi-task-learning.py]: Looks nice! Thanks for catching this. Perhaps we can put it on more lines for easier readability:

```suggestion
Multi task learning is achieved quite easily by initializing the SentenceTransformerTrainer like this:
SentenceTransformerTrainer(
    train_dataset={"mnrl": MultipleNegativesRankingLoss_train_dataset, "cl": ContrastiveLoss_train_dataset},
    loss={"mnrl": MultipleNegativesRankingLoss(model=model), "cl": ContrastiveLoss(model=model)},
    ...
)
```

**Comment by harshitsharma496** [comment]
@tomaarsen PTAL, Thnx

**Comment by tomaarsen** [comment]
Looks great! This'll be merged automatically when the tests pass.
